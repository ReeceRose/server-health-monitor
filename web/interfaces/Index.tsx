// Generic types

export interface StandardResponse {
  Data: unknown;
  StatusCode: number;
  Error: string;
  Success: boolean;
}

export interface Host {
  hostname?: string;
  uptime?: number;
  bootTime?: number;
  procs?: number;
  os?: string;
  platform?: string;
  platformFamily?: string;
  platformVersion?: string;
  kernelVersion?: string;
  kernelArch?: string;
  virtualizationSystem?: string;
  virtualizationRole?: string;
  hostID?: string;
  online?: boolean;
}

export interface Health {
  id: string;
  agentID: string;
  createTime: number;
  updateTime: number;
  online?: boolean;
}

export interface HealthResponse extends StandardResponse {
  Data: Health[];
}

export interface HostResponse extends StandardResponse {
  Data: Host[];
}
