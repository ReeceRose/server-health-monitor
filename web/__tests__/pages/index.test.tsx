import { render } from '@testing-library/react';
import { Health } from '../../interfaces/Index';

import Home from '../../pages/index';

const health: Health[] = [
  {
    agentID: '1',
    createTime: 1,
    updateTime: 1,
    id: '1',
    online: true,
    host: {
      os: 'ubunut',
      platform: 'linux',
      hostname: 'Test Machine',
    },
  },
];

test('Index page matches snapshot', () => {
  const { asFragment } = render(<Home error="" health={health} />);
  expect(asFragment()).toMatchSnapshot();
});
