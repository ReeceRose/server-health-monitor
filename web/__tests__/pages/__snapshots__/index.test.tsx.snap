// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="relative pt-12 pb-32 bg-gray-800 md:pt-32"
  >
    <div
      class="w-full px-4 mx-auto md:px-10"
    >
      <div>
        <div
          class="flex flex-wrap"
        >
          <div
            class="w-full px-4 lg:w-6/12 xl:w-3/12"
          >
            <div
              class="relative flex flex-col min-w-0 mb-6 break-words bg-white rounded shadow-lg xl:mb-0"
            >
              <div
                class="flex-auto p-4"
              >
                <div
                  class="flex flex-wrap"
                >
                  <div
                    class="relative flex-1 flex-grow w-full max-w-full pr-4"
                  >
                    <h5
                      class="text-xs font-bold text-gray-400 uppercase"
                    >
                      ACTIVE SERVERS
                    </h5>
                    <span
                      class="text-xl font-semibold text-gray-700"
                    >
                      0
                    </span>
                  </div>
                  <div
                    class="relative flex-initial w-auto pl-4"
                  >
                    <div
                      class="text-white p-3 text-center inline-flex items-center justify-center w-12 h-12 shadow-lg rounded-full bg-green-400"
                    >
                      <i
                        class="fas fa-server"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="w-full px-4 lg:w-6/12 xl:w-3/12"
          >
            <div
              class="relative flex flex-col min-w-0 mb-6 break-words bg-white rounded shadow-lg xl:mb-0"
            >
              <div
                class="flex-auto p-4"
              >
                <div
                  class="flex flex-wrap"
                >
                  <div
                    class="relative flex-1 flex-grow w-full max-w-full pr-4"
                  >
                    <h5
                      class="text-xs font-bold text-gray-400 uppercase"
                    >
                      INACTIVE SERVERS
                    </h5>
                    <span
                      class="text-xl font-semibold text-gray-700"
                    >
                      1
                    </span>
                  </div>
                  <div
                    class="relative flex-initial w-auto pl-4"
                  >
                    <div
                      class="text-white p-3 text-center inline-flex items-center justify-center w-12 h-12 shadow-lg rounded-full bg-red-400"
                    >
                      <i
                        class="fas fa-server"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="flex-grow w-full px-4 mx-auto mb-0 -m-24 md:px-10"
  >
    <div
      class="flex flex-wrap"
    >
      <div
        class="w-full px-4 mb-12 xl:mb-0"
      >
        <div
          class="relative flex flex-col break-words w-full mb-6 shadow-lg rounded bg-gray-700 text-white"
        >
          <div
            class="px-4 py-3 mb-0 border-0 rounded-t"
          >
            <div
              class="flex flex-wrap items-center"
            >
              <div
                class="relative flex-1 flex-grow w-full max-w-full px-2"
              >
                <h3
                  class="font-semibold text-lg text-white"
                >
                  Servers
                </h3>
              </div>
            </div>
          </div>
          <div
            class="block w-full overflow-x-auto"
          >
            <table
              class="items-center w-full bg-transparent border-collapse"
            >
              <thead>
                <tr>
                  <th
                    class="px-6 align-middle border border-solid py-3 text-xs uppercase border-l-0 border-r-0 whitespace-nowrap font-semibold text-left bg-gray-600 text-gray-200 border-gray-500"
                  >
                    Hostname
                  </th>
                  <th
                    class="px-6 align-middle border border-solid py-3 text-xs uppercase border-l-0 border-r-0 whitespace-nowrap font-semibold text-left bg-gray-600 text-gray-200 border-gray-500"
                  >
                    OS
                  </th>
                  <th
                    class="px-6 align-middle border border-solid py-3 text-xs uppercase border-l-0 border-r-0 whitespace-nowrap font-semibold text-left bg-gray-600 text-gray-200 border-gray-500"
                  >
                    Platform
                  </th>
                  <th
                    class="px-6 align-middle border border-solid py-3 text-xs uppercase border-l-0 border-r-0 whitespace-nowrap font-semibold text-left bg-gray-600 text-gray-200 border-gray-500"
                  >
                    Last Ping
                  </th>
                  <th
                    class="px-6 align-middle border border-solid py-3 text-xs uppercase border-l-0 border-r-0 whitespace-nowrap font-semibold text-left bg-gray-600 text-gray-200 border-gray-500"
                  >
                    Status
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <th
                    class="flex items-center p-4 px-6 text-xs text-left align-middle border-t-0 border-l-0 border-r-0 whitespace-nowrap"
                  >
                    <span
                      class="text-white"
                    >
                      Test Machine
                    </span>
                  </th>
                  <td
                    class="p-4 px-6 text-xs align-middle border-t-0 border-l-0 border-r-0 whitespace-nowrap"
                  >
                    <span
                      class="text-white"
                    >
                      ubunut
                    </span>
                  </td>
                  <td
                    class="p-4 px-6 text-xs align-middle border-t-0 border-l-0 border-r-0 whitespace-nowrap"
                  >
                    <span
                      class="text-white"
                    >
                      linux
                    </span>
                  </td>
                  <td
                    class="p-4 px-6 text-xs align-middle border-t-0 border-l-0 border-r-0 whitespace-nowrap"
                  >
                    TODO
                  </td>
                  <td
                    class="p-4 px-6 text-xs align-middle border-t-0 border-l-0 border-r-0 whitespace-nowrap"
                  >
                    <span>
                      <i
                        class="mr-2 text-red-400 fas fa-circle"
                      />
                       Offline
                    </span>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
